modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Captcha/Captcha.es.js

```diff
@@ -46,6 +46,29 @@ const Captcha = ({html, name}) => {
 		}
 	}, [elRef]);
 
+	useEffect(() => {
+		function setNewCaptcha() {
+			const captcha = elRef.current.querySelector('img');
+			const url = new URL(captcha.src);
+
+			url.searchParams.set('t', Date.now());
+
+			captcha.setAttribute('src', url.href);
+		}
```

Comment:
I declared this method in here to minimize pain with changes in references, let me know if you see a problem with that.

Reply:
Hey @matuzalemsteles! The problem I see there is that the Captcha field doesn't actually know what implementation of captcha might be configured in portal. So the method above will only be valid when simplecaptcha is configured. But it would be best if we were agnostic from the implementation.

About the problems with SimpleCaptcha and senna it sounds like something we need to fix separately because other places that might use captcha might suffer from the same problem right now.

Reply:
> Hey @matuzalemsteles! The problem I see there is that the Captcha field doesn't actually know what implementation of captcha might be configured in portal. So the method above will only be valid when simplecaptcha is configured. But it would be best if we were agnostic from the implementation.

Hmm, yeah, you're right. I don't think that should be added here. I'm thinking maybe to add `Liferay.fire` on the React side and `Liferay.on` on the `simplecaptcha.jsp` side so that he can add the events but still maintaining the compatibility he has today.

> About the problems with SimpleCaptcha and senna it sounds like something we need to fix separately because other places that might use captcha might suffer from the same problem right now.

In fact the problems with Senna.js is a peculiar case on our side, why we need to wait for React to finish rendering, in the navigations with Senna.js we don't have events that are triggered when this happens most of our events happen before it starts to render, so we don't know when to add the events on the `simplecaptcha.jsp` side.

Another problem is when the portlet is loaded and there is a Captcha, it currently works well since it will load the `simplecaptcha.jsp` script but we return to the same problem of waiting for React to render to add the events.